src/collection.rs

```diff
@@ -23,6 +26,163 @@ use strategy::*;
+    /// + `max <= self.len()`, since the max size can never happen.
+    /// + `min >= max`, since the interval is empty.
+    ///
+    /// This is equivalent to a panic ensuing iff: `!(min <= len < max)`.
+    fn prop_sample(self, size: Range<usize>) -> Self::Strategy;
+}
+
+impl<T: Clone + fmt::Debug> CollectionStrategy for Vec<T> {
+    type Strategy = SubseqStrategy<T>;
+    type ValueTree = SubseqValueTree<T>;
+    fn prop_sample(self, size: Range<usize>) -> Self::Strategy {
+        SubseqStrategy::new(self, size)
+    }
+}
+
+/// A `Strategy` to generate subsequences, of a `seq: Vec<T>`,
+/// where the subsequences have a size: `min <= seq.len() < max`.
+#[derive(Debug, Clone)]
+pub struct SubseqStrategy<T: Clone + fmt::Debug> {
+    full: Vec<T>,
+    size: Range<usize>,
+}
+
+impl<T: Clone + fmt::Debug> SubseqStrategy<T> {
+    /// Constructs a `SubseqStrategy` which will generate subsequences of
+    /// `source` where the subsequences have a number of elements `len` for
+    /// which `min <= len < max` always holds where `size = `min..max`.
+    ///
+    /// # Safety
+    ///
+    /// If any of the following hold, a panic will ensue:
+    ///
+    /// + `min > self.len()`, since there are fewer elements than the minimum.
+    /// + `max <= self.len()`, since the max size can never happen.
+    /// + `min >= max`, since the interval is empty.
+    ///
+    /// This is equivalent to a panic ensuing iff: `!(min <= len < max)`.
+    pub fn new(source: Vec<T>, size: Range<usize>) -> Self {
+        let len = source.len();
+        assert!(size.start <= len);
+        assert!(len < size.end);
+        Self {
+            full: source,
+            size: size,
+        }
+    }
+}
+
+impl<T: Clone + fmt::Debug> Strategy for SubseqStrategy<T> {
+    type Value = SubseqValueTree<T>;
+    
+    fn new_value(&self, runner: &mut TestRunner)
+        -> Result<Self::Value, String>
+    {
+        // Generate the amount of elements in result:
+        let rng = runner.rng();
+        let amount = rand::distributions::Range::new(
+            self.size.start, self.size.end).ind_sample(rng);
+
+        // Generate a subsequence of amount size.
+        let subvec = rand::sample(rng, self.full.iter().cloned(), amount);
+
+        // We're done, yield the value tree.
+        Ok(SubseqValueTree {
+            data: subvec,
+            len: amount,
+            min: self.size.start,
+        })
+    }
+}
+
+/// The `ValueTree` for `SubseqStrategy`.
+pub struct SubseqValueTree<T: Clone + fmt::Debug> {
+    data: Vec<T>,
+    len: usize,
+    min: usize
+}
+
+impl<T: Clone + fmt::Debug> ValueTree for SubseqValueTree<T> {
+    type Value = Vec<T>;
+
+    fn current(&self) -> Self::Value {
+        // We verify manually that self.len <= self.data.len(),
+        // therefore it is safe. The assertion should always hold.
+        assert!(self.len <= self.data.len());
+        unsafe {
+            slice::from_raw_parts(self.data.as_ptr(), self.len)
```

Comment:
Why not `self.data[..self.len].into()`?

Reply:
Oh yeah, you can (and should) do that :P
You can take over from here ;)
I guess you can change this in additional commits.